File: daily/Arrays/3541.Find-most-frequent-vowel-and-consonant.go
package daily

func maxFreqSum(s string) int {
	maxVowel, maxConsonant := 0, 0
	map_Vowel, map_consonant := make(map[rune]int), make(map[rune]int)

	for _, c := range s {
		if c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' {
			map_Vowel[rune(c)]++
			maxVowel = max(maxVowel, map_Vowel[rune(c)])
		} else {
			map_consonant[rune(c)]++
			maxConsonant = max(maxConsonant, map_consonant[rune(c)])
		}
	}

	return maxVowel + maxConsonant
}

File: daily/Arrays/3541.Find-most-frequent-vowel-and-consonant.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        vowels = set("aeiou")
        max_vowel = 0
        max_consonant = 0
        vcount = {}
        ccount = {}

        for ch in s:
            if ch in vowels:
                vcount[ch] = vcount.get(ch, 0) + 1
                if vcount[ch] > max_vowel:
                    max_vowel = vcount[ch]
            else:
                ccount[ch] = ccount.get(ch, 0) + 1
                if ccount[ch] > max_consonant:
                    max_consonant = ccount[ch]

        return max_vowel + max_consonant

